src/xunit.performance.api/Profilers/Etw/SimpleTraceEventParser.cs

```diff
@@ -0,0 +1,378 @@
+                        process.LifeSpan.Start = obj.TimeStamp;
+                        processes.Add(process);
+                    }
+                };
+                parser.ProcessStop += (ProcessTraceData obj) => {
+                    if (IsOurProcess(obj))
+                        processes.Single(process => process.Id == obj.ProcessID).LifeSpan.End = obj.TimeStamp;
+                };
+
+                ////////////////////////////////////////////////////////////////
+                // Image/Module data
+                parser.ImageLoad += (ImageLoadTraceData obj) => {
+                    var process = processes.SingleOrDefault(p => p.Id == obj.ProcessID);
+                    if (process == null)
+                        return;
+
+                    var module = new Module(obj.FileName, obj.ImageChecksum) {
+                        AddressSpace = new AddressSpace(obj.ImageBase, (uint)obj.ImageSize)
+                    };
+                    module.LifeSpan.Start = obj.TimeStamp;
+
+                    process.Modules.Add(module);
+                };
+                parser.ImageUnload += (ImageLoadTraceData obj) => {
+                    var process = processes.SingleOrDefault(p => p.Id == obj.ProcessID);
+                    if (process == null)
+                        return;
+
+                    // Check if the unloaded module is on the list.
+                    // The module must be loaded, in the same address space, and same file name.
+                    var module = process.Modules
+                        .SingleOrDefault(m => obj.TimeStamp > m.LifeSpan.End && obj.Equal(m));
+                    if (module == null)
+                        return;
+                    module.LifeSpan.End = obj.TimeStamp;
+                };
+
+                ////////////////////////////////////////////////////////////////
+                // "ntoskrnl.exe" data
+                Func<ImageLoadTraceData, bool> IsNtoskrnlModule = (ImageLoadTraceData obj) => {
+                    return obj.ProcessID == 0 && obj.FileName.EndsWith("ntoskrnl.exe");
+                };
+                parser.ImageDCStart += (ImageLoadTraceData obj) => {
+                    if (IsNtoskrnlModule(obj))
+                    {
+                        if (defaultNtoskrnlModule == null)
+                            defaultNtoskrnlModule = new Module(obj.FileName, obj.ImageChecksum);
+
+                        defaultNtoskrnlModule.AddressSpace = new AddressSpace(obj.ImageBase, (uint)obj.ImageSize);
+                        defaultNtoskrnlModule.LifeSpan.Start = obj.TimeStamp;
+                        defaultNtoskrnlModule.LifeSpan.End = DateTime.MaxValue;
+                    }
+                };
+                parser.ImageDCStop += (ImageLoadTraceData obj) => {
+                    if (IsNtoskrnlModule(obj)
+                        && defaultNtoskrnlModule != null
```

Comment:
You probably don't need to process ImageDCStop events.  DCStop events are fired at the end of the process or trace so you are guaranteed that the module is loaded until the end of the process.